src/Test/UnitTests/constraints.jl

```diff
@@ -22,3 +22,143 @@ function getconstraint(model::MOI.ModelLike, config::TestConfig)
     @test MOI.isvalid(model, c2)
 end
 unittests["getconstraint"]    = getconstraint
+
+function testsolution(model, config;
+        objective_value   = nothing,
+        variable_primal   = nothing,
+        constraint_primal = nothing,
+        constraint_dual   = nothing
+    )
+    atol, rtol = config.atol, config.rtol
+    MOI.optimize!(model)
+    @test MOI.canget(model, MOI.TerminationStatus())
+    @test MOI.get(model, MOI.TerminationStatus()) == MOI.Success
+    @test MOI.canget(model, MOI.PrimalStatus())
+    @test MOI.get(model, MOI.PrimalStatus()) == MOI.FeasiblePoint
+    if objective_value != nothing
+        @test MOI.canget(model, MOI.ObjectiveValue())
+        @test MOI.get(model, MOI.ObjectiveValue()) ≈ objective_value atol=atol rtol=rtol
+    end
+    if variable_primal != nothing
+        for (index, solution_value) in variable_primal
+            @test MOI.canget(model, MOI.VariablePrimal(), MOI.VariableIndex)
+            @test MOI.get(model, MOI.VariablePrimal(), index) ≈ solution_value atol=atol rtol=rtol
+        end
+    end
+    if constraint_primal != nothing
+        for (index, solution_value) in constraint_primal
+            @test MOI.canget(model, MOI.ConstraintPrimal(), typeof(index))
+            @test MOI.get(model, MOI.ConstraintPrimal(), index) ≈ solution_value atol=atol rtol=rtol
+        end
+    end
+    if constraint_dual != nothing
+        if config.duals
```

Comment:
It's worth documenting that the constraint dual isn't tested unless `config.duals == true`, because that's surprising.